src/wp-includes/general-template.php

```diff
@@ -4746,7 +4746,7 @@ function paginate_links( $args = '' ) {
 					$link = add_query_arg( $add_args, $link );
 				}
 				$link .= $args['add_fragment'];
-				$link  = get_option( 'permalink_structure' ) ? user_trailingslashit( $link, 'paged' ) : $link;
+				$link  = get_option( 'permalink_structure' ) && ! is_search() ? user_trailingslashit( $link, 'paged' ) : $link;
```

Comment:
As discussed on the ticket, I think it would be preferable to check if there is a query string parameter in the URL rather than limit it to search. That will ensure that plugins making use of query strings aren't affected by the same bug.

The same should probably be done for URL fragments, so I've included that here too.

```suggestion
				// 
				$link  = get_option( 'permalink_structure' ) && ! str_contains( $link, '?' ) && ! str_contains( $link, '#' ) ? user_trailingslashit( $link, 'paged' ) : $link;
```

(WordPress includes a polyfill for `str_contains()` so it can be used without worrying about older versions of PHP)

This will also need to be added in the other locations in the original patch, see https://core.trac.wordpress.org/changeset/59966

If you have any experience writing unit tests, it would be great if you can add some for search URLs. You can see the existing tests on the link above. I'm happy to help if you are unsure how to write the tests.